compiler/rustc_codegen_llvm/src/gotoc/mod.rs

```diff
@@ -305,18 +379,29 @@ impl CodegenBackend for GotocCodegenBackend {
                 }
             }
         }
+        clear_panic_debug_codegen_item();
 
         // then we move on to codegen
         for cgu in codegen_units {
             let items = cgu.items_in_deterministic_order(tcx);
             for (item, _) in items {
                 match item {
```

Comment:
Moving/adding the clear as the first item in the loop would make it clear that we're doing clear when needed :)

Reply:
Changed to implement as a closure that sets/clears the value as appropriate!